idaes/core/util/tables.py

```diff
@@ -266,22 +268,50 @@ def stream_table_dataframe_to_string(stream_table, **kwargs):
         na_rep=na_rep, justify=justify, float_format=float_format, **kwargs
     )
 
-
-def _get_state_from_port(port, time_point):
-    # Check port for _state_block attribute
-    try:
-        if len(port._state_block) == 1:
-            return port._state_block[0][time_point]
-        else:
-            return port._state_block[0][time_point, port._state_block[1]]
-    except AttributeError:
-        # Port was not created by IDAES add_port methods. Return exception for
-        # the user to fix.
-        raise ConfigurationError(
-            f"Port {port.name} does not have a _state_block attribute, "
-            f"thus cannot determine StateBlock to use for collecting data."
-            f" Please provide the associated StateBlock instead, or use "
-            f"the IDAES add_port methods to create the Port."
+def _get_state_from_port(port,time_point):
+    from idaes.core.property_base import StateBlockData
+    states = list()
+    v = None # To make PyLint stop complaining
+    for v in port.iter_vars():
+        if not v.parent_block().parent_component().is_indexed():
+            raise TypeError(
+                f"No state block could be retrieved from Port {port.name} "
+                f"because variable {v.name} does not belong to a state block."
```

Comment:
First, the error message here needs to be updated to be more clear about the specific issue - i.e. that the parent component is expected to be indexed.

Second, whilst this condition should be true for any core IDAES model, it might be a bit restrictive for special cases (e.g. surrogate models, custom codes models). Of course, allowing for this would mean having to catch that possibility later on too.

Reply:
If the parent component is indexed, then yes we need to deal with that and return the correct IndexedBlock. However, I am not sure that it is necessary to demand that the parent Block *always* be indexed. I.e. I think we should support the case where the parent happens to be a SimpleBlock as well. We do not want to return a slicer however.

Also note that many Pyomo methods only work in the IndexedBlock or SimpleBlock, and do not exist on BlockData objects (and sometime the reverse), so it is up to the tool looking for those methods to decide how to handle that.

Reply:
The function no longer checks whether the parent component is indexed, but just tries to retrieve an index. It appears even unindexed Pyomo objects have an index of None, which can then be used to index them, so the function will probably work on `SimpleBlock`s. If you could direct me to a particular example, I would be happy to write a test to ensure this functionality.